tests/solidity_test_helpers/MockUniverse.sol

```diff
@@ -0,0 +1,315 @@
+        return setTargetRepMarketCapInAttoethValue;
+    }
+    function getValidityBond() public returns (uint256) {
+        return setValidityBondValue;
+    }
+    function getDesignatedReportStake() public returns (uint256) {
+        return setDesignatedReportStakeValue;
+    }
+    function getDesignatedReportNoShowBond() public returns (uint256) {
+        return setDesignatedReportNoShowBondValue;
+    }
+    function getReportingFeeInAttoethPerEth() public returns (uint256) {
+        return setReportingFeeInAttoethPerEthValue;
+    }
+    function getRepAvailableForExtraBondPayouts() public view returns (uint256) {
+        return setRepAvailableForExtraBondPayoutsValue;
+    }
+    function increaseRepAvailableForExtraBondPayouts(uint256 _amount) public returns (bool) {
+        return setIncreaseRepAvailableForExtraBondPayoutsValue;
+    }
+    function decreaseRepAvailableForExtraBondPayouts(uint256 _amount) public returns (bool) {
+        return setDecreaseRepAvailableForExtraBondPayoutsValue;
+    }
+    function getExtraDisputeBondRemainingToBePaidOut() public view returns (uint256) {
+        return setExtraDisputeBondRemainingToBePaidOutValue;
+    }
+    function increaseExtraDisputeBondRemainingToBePaidOut(uint256 _amount) public returns (bool) {
+        return setIncreaseExtraDisputeBondRemainingToBePaidOutValue;
+    }
+    function decreaseExtraDisputeBondRemainingToBePaidOut(uint256 _amount) public returns (bool) {
+        return setDecreaseExtraDisputeBondRemainingToBePaidOutValue;
+    }
+    function calculateFloatingValue(uint256 _badMarkets, uint256 _totalMarkets, uint256 _targetDivisor, uint256 _previousValue, uint256 _defaultValue, uint256 _floor) public pure returns (uint256 _newValue) {
+        if (_totalMarkets == 0) {
+            return _defaultValue;
+        }
+        if (_previousValue == 0) {
+            _previousValue = _defaultValue;
+        }
+
+        // Modify the amount based on the previous amount and the number of markets fitting the failure criteria. We want the amount to be somewhere in the range of 0.5 to 2 times its previous value where ALL markets with the condition results in 2x and 0 results in 0.5x.
+        if (_badMarkets <= _totalMarkets / _targetDivisor) {
+            // FXP formula: previous_amount * actual_percent / (2 * target_percent) + 0.5;
+            _newValue = _badMarkets.mul(_previousValue).mul(_targetDivisor).div(_totalMarkets).div(2)  + _previousValue / 2; // FIXME: This is on one line due to solium bugs
+        } else {
+            // FXP formula: previous_amount * (1/(1 - target_percent)) * (actual_percent - target_percent) + 1;
+            _newValue = _targetDivisor.mul(_previousValue.mul(_badMarkets).div(_totalMarkets).sub(_previousValue.div(_targetDivisor))).div(_targetDivisor - 1) + _previousValue; // FIXME: This is on one line due to a solium bug
+        }
+
+        if (_newValue < _floor) {
+            _newValue = _floor;
+        }
```

Comment:
Why not just have this return a previously set value like all of the other getters?

Reply:
I had an issue with the "pure" compile/upload contract failed. So I just copied the function from universe.

Reply:
Ah, yeah.  If this method is marked as pure the you would have to hard-code a value.  This is fine.